automation/tdx_canonical_verifier.sh

```diff
@@ -0,0 +1,184 @@
+        set -x
+        output=$(ls -l /dev/sgx_*)
+        set +x
+        if [[ $output =~ "/dev/sgx_enclave" && $output =~ "/dev/sgx_provision" && $output =~ "/dev/sgx_vepc" ]]; then
+                        echo -e "SGX enabled and devices present.\n"
+        else
+                        echo -e "Failure: SGX not enabled in BIOS, missing SGX devices. Exiting..!\n"
+                        exit 1
+        fi
+}
+
+verify_service_status(){
+        echo -e "$1"
+        if ! [[ "$1" =~ "$SERVICE_ACTIVE_STATUS_VERIFY_TEXT" ]]; then
+                echo -e "\nFailure: attestation Service is not active. Please verify...\n"
+                exit 1
+        fi
+}
+
+verify_attestation_services() {
+        echo -e "\n\nVerifying Attestation services on the host...\n"
+        status=$(sudo systemctl status qgsd)
+        verify_service_status "$status"
+
+        status=$(sudo systemctl status pccs)
+        verify_service_status "$status"
+
+        sudo rm -rf /var/log/mpa_registration.log
+        sudo systemctl restart mpa_registration_tool
+        sleep 5
+        log=$(cat /var/log/mpa_registration.log)
+        echo "$log"
+        if [[ "$log" =~ "$MPA_REGISTRATION_CHECK" ]]; then
+                echo "MPA registration is successful"
+        else
+                echo "ERROR : MPA registration failure"
+                exit 1
+        fi
+
+        status=$(sudo systemctl status mpa_registration_tool)
+        echo -e "$status"
+        if ! [[ "$status" =~ "$MPA_SERVICE_CHECK" ]]; then
+                echo "\nFailure: MPA Service is not enabled. Please verify...\n"
+                exit 1
+        fi
+}
+
+verify_attestation_host(){
+        verify_sgx_devices
+        verify_attestation_services
+}
+
+verify_attestation_guest(){
+        echo -e "\n\nVerifying Attestation services on the guest...\n"
+        cd "$CUR_DIR"
+        sshpass -p "$TD_GUEST_PASSWORD" scp -vv -P "$TD_GUEST_PORT" -r "$TDX_DIR" "$VERIFY_ATTESTATION_SCRIPT" "$TRUSTAUTHORITY_API_FILE" root@localhost:/tmp/ 2>&1 || (echo "scp command failure" && exit 1)
+        sshpass -p "$TD_GUEST_PASSWORD" ssh -T -o StrictHostKeyChecking=no -p "$TD_GUEST_PORT" root@localhost "cd /tmp; ./$VERIFY_ATTESTATION_SCRIPT" 2>&1 /dev/tty || (echo "ERROR: attestation verification error on td guest" && exit 1)
+}
+
+args=$(getopt -a -o i:h -l td_image_path: -- "$@")
+if [[ $? -gt 0 ]]; then
+  usage
+fi
+
+eval set -- "$args"
+while :
+do
+        case $1 in
+                -i | --td_image_path) TD_IMG=$2 ; break ;;
+                -h) usage; shift ;;
+                --) shift; break ;;
+                *) >&2 echo Unsupported option: $1
+                        usage ;;
+        esac
+done
+
+if [[ -z $TD_IMG ]]; then
+        echo -e "ERROR: -i or --td_image_path is mandatory argument.";
+        exit 1;
+fi
+
+apt install --yes sshpass &> /dev/null
+clone_tdx_repo
+verify_tdx_host
+check_production_system
+verify_attestation_host

```

Comment:
Only call attestation if its a production system, so check the value from check_production and take action accordingly.

Reply:
Since we clone tdx repo again during verifier script call. we need to check the system status